src/auth/computeclient.ts

```diff
@@ -87,27 +87,20 @@ export class Compute extends OAuth2Client {
     return super.requestAsync<T>(opts, retry).catch(e => {
       const res = (e as GaxiosError).response;
       if (res && res.status) {
-        let helpfulMessage = null;
-        if (res.status === 403) {
-          helpfulMessage =
-              'A Forbidden error was returned while attempting to retrieve an access ' +
-              'token for the Compute Engine built-in service account. This may be because the Compute ' +
-              'Engine instance does not have the correct permission scopes specified.';
-        } else if (res.status === 404) {
-          helpfulMessage =
-              'A Not Found error was returned while attempting to retrieve an access' +
-              'token for the Compute Engine built-in service account. This may be because the Compute ' +
-              'Engine instance does not have any permission scopes specified.';
-        }
-        if (helpfulMessage) {
-          if (e && e.message && !retry) {
-            helpfulMessage += ' ' + e.message;
-          }
-          if (e) {
-            e.message = helpfulMessage;
-          } else {
-            e = new Error(helpfulMessage);
-            (e as NodeJS.ErrnoException).code = res.status.toString();
+        (e as NodeJS.ErrnoException).code = res.status.toString();
+        if (!retry) {
+          if (res.status === 403) {
+            e.message =
+                'A Forbidden error was returned while attempting to retrieve an access ' +
+                'token for the Compute Engine built-in service account. This may be because the Compute ' +
+                'Engine instance does not have the correct permission scopes specified: ' +
+                e.message;
+          } else if (res.status === 404) {
+            e.message =
+                'A Not Found error was returned while attempting to retrieve an access' +
+                'token for the Compute Engine built-in service account. This may be because the Compute ' +
+                'Engine instance does not have any permission scopes specified: ' +
+                e.message;
```

Comment:
I don't get it :) It does make the code look much better, but it does not add anything to the message: both old and new code append `e.message`, it has not change. In the users' example the message they got was

```
Forbidden error was returned while attempting to retrieve an access token for the Compute Engine built-in service account. This may be because the Compute Engine instance does not have the correct permission scopes specified. The caller does not have permission
```

which clearly has `e.message` appended: `The caller does not have permission`

Reply:
You ...  are totally right.  One of the commenters on the bug had a great hidden gem:

> It looks like https://github.com/googleapis/google-auth-library-nodejs/blob/master/src/auth/computeclient.ts#L92 is potentially firing on any 403/404 rather than just ones related to getting a token?

We were appending that message for any *request*, not just requests to get the access token.  Helpfully, the unit tests ensured we kept on doing the wrong thing.  

Keep making comments - every time you do it I get to delete more code!